src/coreclr/jit/codegenarmarch.cpp

```diff
@@ -2416,7 +2416,7 @@ class CopyBlockUnrollHelper
 
         const int dstOffsetAligned = AlignUp((UINT)dstOffset, storePairRegsAlignment);
 
-        if (endDstOffset - dstOffsetAligned >= storePairRegsWritesBytes)
+        if (byteCount >= (unsigned)storePairRegsWritesBytes)
```

Comment:
`UnrollInitBlock` has somewhat similar logic. Does it need to be updated?

Reply:
No - there is only one `dstOffset` in InitBlock case. 

If a store instruction couldn't be encoded during initial `TryEncodeAllOffsets` call - we spill the whole `dstAddr` to a register (effectively, setting `dstOffset` to `0`). In other words, there is only one *"degree of freedom"* for InitBlock while for CopyBlock we have two (and in this case, changing value of one `dstOffset` caused going from a state `{canEncodeAllLoads: true, canEncodeAllStores: false}` to `{canEncodeAllLoads: false, canEncodeAllStores: true}`)